lib/vrl/stdlib/src/parse_aws_clb_log.rs

```diff
@@ -0,0 +1,332 @@
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let bytes = self.value.resolve(ctx)?;
+        parse_aws_clb_log(bytes)
+    }
+
+    fn type_def(&self, _: &state::TypeState) -> TypeDef {
+        TypeDef::object(inner_kind()).fallible(/* log parsing error */)
+    }
+}
+
+fn inner_kind() -> BTreeMap<Field, Kind> {
+    BTreeMap::from([
+        (
+            Field::from("backend_processing_time"),
+            Kind::bytes() | Kind::null(),
+        ),
+        (
+            Field::from("backend_status_code"),
+            Kind::bytes() | Kind::null(),
+        ),
+        (Field::from("client_host"), Kind::bytes()),
+        (Field::from("elb"), Kind::bytes()),
+        (Field::from("elb_status_code"), Kind::bytes()),
+        (Field::from("received_bytes"), Kind::integer()),
+        (Field::from("request_method"), Kind::bytes() | Kind::null()),
+        (Field::from("request_processing_time"), Kind::float()),
+        (
+            Field::from("request_protocol"),
+            Kind::bytes() | Kind::null(),
+        ),
+        (Field::from("request_url"), Kind::bytes() | Kind::null()),
+        (Field::from("response_processing_time"), Kind::float()),
+        (Field::from("sent_bytes"), Kind::integer()),
+        (Field::from("ssl_cipher"), Kind::bytes() | Kind::null()),
+        (Field::from("ssl_protocol"), Kind::bytes() | Kind::null()),
+        (Field::from("target_host"), Kind::bytes() | Kind::null()),
+        (Field::from("time"), Kind::bytes()),
+        (Field::from("user_agent"), Kind::bytes() | Kind::null()),
+    ])
+}
+
+fn parse_log(mut input: &str) -> Result<Value> {
+    let mut log = BTreeMap::new();
+
+    macro_rules! get_value {
+        ($name:expr, $parser:expr) => {{
+            let result: IResult<&str, _, (&str, nom::error::ErrorKind)> = $parser(input);
+            match result {
+                Ok((rest, value)) => {
+                    input = rest;
+                    value
+                }
+                Err(error) => {
+                    return Err(format!("failed to get field `{}`: {}", $name, error).into())
+                }
+            }
+        }};
+    }
+    macro_rules! field_raw {
+        ($name:expr, $parser:expr) => {
+            log.insert(
+                $name.into(),
+                match get_value!($name, $parser).into() {
+                    Value::Bytes(bytes) if bytes == "-" => Value::Null,
+                    value => value,
+                },
+            )
+        };
+    }
+    macro_rules! field {
+        ($name:expr, $($pattern:pat_param)|+) => {
+            field_raw!($name, preceded(char(' '), take_while1(|c| matches!(c, $($pattern)|+))))
+        };
+    }
+    macro_rules! field_parse {
+        ($name:expr, $($pattern:pat_param)|+, $type:ty) => {
+            field_raw!($name, map_res(preceded(char(' '), take_while1(|c| matches!(c, $($pattern)|+))), |s: &str| s.parse::<$type>()))
+        };
+    }
```

Comment:
These macro rules appear to be the same as those in `parse_aws_alb_log`, would it be possible to move those into a shared file and reuse them here?